src/Console/Test/console_main.cpp

```diff
@@ -35,18 +33,47 @@ QWidget* mainWindowBuilder(FOEDAG::CommandLine* cmd,
 }
 
 void registerExampleCommands(FOEDAG::Session* session) {
-  auto console_open = [](void* clientData, Tcl_Interp* interp, int argc,
+  auto console_pwd = [](void* clientData, Tcl_Interp* interp, int argc,
+                        const char* argv[]) -> int {
+    Q_UNUSED(clientData)
+    Q_UNUSED(argv)
+    Q_UNUSED(argc)
+    FOEDAG::TclConsoleWidget* console = FOEDAG::Init(interp);
+    QString res = console->getPrompt() + "pwd\n" + QDir::currentPath() + "\n" +
+                  console->getPrompt();
+    CHECK_EXPECTED("pwd", res)
+    return 0;
+  };
+  session->TclInterp()->registerCmd("console_pwd", console_pwd,
+                                    GlobalSession->TclInterp(), nullptr);
+
+  auto console_proc = [](void* clientData, Tcl_Interp* interp, int argc,
                          const char* argv[]) -> int {
     Q_UNUSED(clientData)
     Q_UNUSED(argv)
     Q_UNUSED(argc)
-    auto console = new FOEDAG::TclConsoleWidget{
-        interp, std::make_unique<FOEDAG::TclConsole>(interp, std::cout),
-        new FOEDAG::StreamBuffer};
-    console->show();
+    FOEDAG::TclConsoleWidget* console = FOEDAG::Init(interp);
+    QString res = console->getPrompt() +
```

Comment:
Hardcoded path, pleas change to an argument and put the path in the tcl test

Reply:
Done